examples/indented_tree.py

```diff
@@ -3,28 +3,34 @@
 ===================
 
 A demonstration of parsing indentation (“whitespace significant” language)
-and the usage of the Indenter class.
+and the usage of the `Indenter` class.
 
 Since indentation is context-sensitive, a postlex stage is introduced to
-manufacture INDENT/DEDENT tokens.
+manufacture `INDENT`/`DEDENT` tokens.
 
-It is crucial for the indenter that the NL_type matches
-the spaces (and tabs) after the newline.
+It is crucial for the indenter that the `NL_type` matches the spaces (and
+tabs) after the newline.
+
+If your whitespace-significant grammar supports comments, then `NL_type`
+must match those comments too. Otherwise, comments that appear in the middle
+of a line will [confuse Lark][1].
+
+[1]: https://github.com/lark-parser/lark/issues/863
 """
 from lark import Lark
 from lark.indenter import Indenter
 
 tree_grammar = r"""
-    ?start: _NL* tree
-
-    tree: NAME _NL [_INDENT tree+ _DEDENT]
-
     %import common.CNAME -> NAME
     %import common.WS_INLINE
-    %declare _INDENT _DEDENT
+    %import common.SH_COMMENT
     %ignore WS_INLINE
+    %ignore SH_COMMENT
+    %declare _INDENT _DEDENT
 
-    _NL: /(\r?\n[\t ]*)+/
+    ?start: _NL* tree
+    tree: NAME _NL [_INDENT tree+ _DEDENT]
+    _NL: (/\r?\n[\t ]*/ | SH_COMMENT)+
 """
 
 class TreeIndenter(Indenter):
```

Comment:
Btw, this is tangential to the purpose of this PR, but I'm curious to know:
1. Why is `tab_len` set to 8? Perhaps I misunderstand its purpose, but the example seems to demonstrate a tab length of 4. And yet, it parses correctly. So I'm a bit confused.
2. These attributes of `Indenter` are decorated with `@property`, so my IDE complains about how they are overridden:
    > "OPEN_PAREN_types" incorrectly overrides property of same name in class "Indenter"
    > Pylance [reportIncompatibleMethodOverride](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportIncompatibleMethodOverride)

    That's perhaps something to improve in the future.

Reply:
Just to be clear, when we say "tab" in this context we mean a literal tab character, correct? i.e. `\t`. So this has nothing to do with specifying how many spaces you want to be part of a single "indent".

Does that mean that when Lark detects indentation (assuming we are only using spaces and not tabs), it doesn't care how many spaces are used? It just looks for the common indentation level.

``` 
a
   b
   c
```

I'm using an indent of 3 spaces here. This will parse fine as an indent and has nothing to do with `tab_len`. 

Am I understanding correctly?

Reply:
Yes, that's right.